packages/evm/src/evm.ts

```diff
@@ -551,7 +570,8 @@ export class EVM implements EVMInterface {
           // in the bytecode of the contract
           if (
             !EOF.validOpcodes(
-              result.returnValue.slice(codeStart, codeStart + eof1CodeAnalysisResults.code)
+              result.returnValue.slice(codeStart, codeStart + eof1CodeAnalysisResults.code),
```

Comment:
I think you mean here that we have somewhat hard-coded either legacy contracts or EOF contracts? I very much agree that this should be loosened, because at this point I am not very happy with it (and that is an understatement :joy: )